numba/caching.py

```diff
@@ -513,3 +530,56 @@ def _save_data(self, name, cres):
 
     def _dump(self, obj):
         return pickle.dumps(obj, protocol=-1)
+
+    def _reduce(self, cres):
+        """
+        Returns a serialized CompileResult
+        """
+        return cres._reduce()
+
+    def _rebuild(self, target_context, payload):
+        """
+        Returns the unserialized CompileResult
+        """
+        return compiler.CompileResult._rebuild(target_context, *payload)
+
+
+class LibraryCache(FunctionCache):
+    """
+    Extends FunctionCache to provide caching of related CodeLibrary objects.
```

Comment:
The description is a bit confusing. Does a LibraryCache work together with another FunctionCache? What are the "other overload entries for the associated function"?